pkg/rbac/namespacedAuthorization.go

```diff
@@ -0,0 +1,68 @@
+// Copyright Contributors to the Open Cluster Management project
+package rbac
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/stolostron/search-v2-api/pkg/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+type userData struct {
+	err        error
+	namespaces []string
+	updatedAt  time.Time
+	lock       sync.Mutex
+}
+
+func (cache *Cache) NamespacedResources(ctx context.Context, clientToken string) (*userData, error) {
+	uid := cache.tokenReviews[clientToken].tokenReview.Status.User.UID //get uid from tokenreview
+
+	cachedUserData, userDataExists := cache.users[uid] //check if userdata cache exists
+	if userDataExists {
+		klog.V(5).Info("Using user data from cache.")
+		return cachedUserData, cachedUserData.err
+
+	}
+
+	// create new instance and
+	user := cache.users[uid]
+	user = &userData{}
```

Comment:
Need a lock around this block to prevent multiple threads from initializing the user at the same time.